### PR TITLE
ELEMENTS-1641: Use feature flag for view vs download

### PR DESCRIPTION
**Pull request by rahuljain-dev**
https://jira.nuxeo.com/browse/ELEMENTS-1641

### CODE REVIEW

**Review by swarnadipa-dev — DISMISSED** [review]

File: core/nuxeo-document.js [hunk 1]
```diff
@@ -186,21 +186,26 @@ import './nuxeo-resource.js';
...
-
-    isCDNEnabled(hostUrl) {
-      return hostUrl.includes('cloudfront.net');
+    isUseDirectUploadS3Enabled() {
+      if (
+        Nuxeo &&
+        Nuxeo.UI &&
+        Nuxeo.UI.config &&
+        Nuxeo.UI.config.s3 &&
+        String(Nuxeo.UI.config.s3.useDirectUpload) === 'true'
```
Comment (rajatgoel-gl) [inline core/nuxeo-document.js]: Kindly optimize this code and remove hardcoded true false if the property is bool type.

File: core/nuxeo-document.js [hunk 2]
```diff
@@ -186,21 +186,26 @@ import './nuxeo-resource.js';
...
         }
       });
     }
 
-    /* This code will be optimized to replace the hardcoded url to use a more generic approach as part of
-        https://jira.nuxeo.com/browse/ELEMENTS-1641 in the upcoming release */
-
-    isCDNEnabled(hostUrl) {
-      return hostUrl.includes('cloudfront.net');
+    isUseDirectUploadS3Enabled() {
```
Comment (rajatgoel-gl) [inline core/nuxeo-document.js]: Use simple and understandable name for the function.

**Review by rajatgoel-gl — CHANGES_REQUESTED** [review]

File: core/nuxeo-document.js (hunk 1 above)
Comment (rahuljain-dev) [reply core/nuxeo-document.js]: done as suggested

File: core/nuxeo-document.js (hunk 2 above)
Comment (rahuljain-dev) [reply core/nuxeo-document.js]: done

**Review by rajatgoel-gl — APPROVED** [review]

**Review by poonamyadav252 — APPROVED** [review]

**Review by rakeshkumar1019 — APPROVED** [review]

**Pull request merged**